raiden_contracts/contracts/SecretRegistry.sol

```diff
@@ -23,7 +23,7 @@ contract SecretRegistry {
     /// @return true if secret was registered, false if the secret was already registered.
     function registerSecret(bytes32 secret) public returns (bool) {
         bytes32 secrethash = keccak256(secret);
-        if (secrethash_to_block[secrethash] > 0) {
+        if (secret == 0x0 || secrethash_to_block[secrethash] > 0) {
```

Comment:
even though this is not part of this PR, just wanna mention we should make this fail with a revert as per [this](https://github.com/raiden-network/spec/issues/76#issuecomment-387373440) issue.